src/compiler/checker.ts

```diff
@@ -201,6 +215,13 @@ namespace ts {
         let assignableRelation: Map<RelationComparisonResult> = {};
         let identityRelation: Map<RelationComparisonResult> = {};
 
+        enum TypeSystemPropertyName {

```

Comment:
Because this is the name of the property you would need to look up in order to discover the type.